journal/src/main/java/io/camunda/zeebe/journal/file/MappedJournalSegmentReader.java

```diff
@@ -100,4 +98,9 @@ public void close() {
   long getNextIndex() {
     return currentIndex + 1;
   }
+
+  private void checkSegmentOpen() {
```

Comment:
:question: Is this still called somewhere?

Reply:
In `seek()`. When seeking the upper layer is supposed to switch to an existing segment. So it call to seek in a deleted segment must never happen.

Reply:
:upside_down_face: If it's only called once, I would just inline it, but that's just a nit.